airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -433,7 +476,24 @@ def _set_name(self, name):
         validate_key(name, max_length=220)
         return re.sub(r'[^a-z0-9.-]+', '-', name.lower())
 
-    def create_pod_request_obj(self) -> k8s.V1Pod:
+    def patch_already_checked(self, pod: k8s.V1Pod):
+        """Add an "already tried annotation to ensure we only retry once"""
```

Comment:
```suggestion
        """Add an "already checked" annotation to ensure we don't reattach on retries"""
```

This might be better? Basically, we're marking this pod as having been cleaned up or handled, and subsequent retries shouldn't try to reattach to it.